src/core/config.c

```diff
@@ -308,18 +308,49 @@ void mCoreConfigPortablePath(char* out, size_t outLength) {
 #endif
 }
 
+void mCoreConfigPortablePath(char* out, size_t outLength) {
+	struct VFile* portableIni;
+	char portableIniPath[PATH_MAX];
+	mCoreConfigPortableIniPath(portableIniPath, sizeof(portableIniPath));
+	out[0] = '\0';
+	if (portableIniPath[0]) {
+		portableIni = VFileOpen(portableIniPath, O_RDONLY);
+		if (portableIni) {
+			// Start with the path that the portable.ini file exists in.
+			if (outLength < PATH_MAX) {
+				char outTmp[PATH_MAX];
```

Comment:
It'd be cleaner (and probably less risky of accidental off-by-one for the null terminator)  if you hosted this and used snprintf in the part you use strncat.